src/rpc/misc.cpp

```diff
@@ -223,7 +224,10 @@ class DescribePaymentAddressVisitor
 public:
     UniValue operator()(const CKeyID &addr) const {
         UniValue obj(UniValue::VOBJ);
-        obj.pushKV("type", "p2pkh");
+        obj.pushKV("address_type", "p2pkh");
+        if (fEnableAddrTypeField) {
```

Comment:
Shouldn't this be called something like `fEnableLegacyAddrTypeField` or `fEnableTypeFieldForAddresses`? The current name is potentially confusing.

Reply:
Consider adding a helper method `pushAddressType(address_type)` to reduce duplicated code.